include/reactor-uc/network_channel.h

```diff
@@ -16,6 +17,7 @@ typedef enum {
   NETWORK_CHANNEL_STATE_CONNECTED,
   NETWORK_CHANNEL_STATE_DISCONNECTED,
   NETWORK_CHANNEL_STATE_LOST_CONNECTION,
+  NETWORK_CHANNEL_STATE_CLOSED,
```

Comment:
This option was missing.
I needed to know if the `CoapChannel` was willingly `closed` on the other federate.
If the channel is closed then a connection to it will always fail.
This is different to `lost_connection` or `disconnected` as the other channel in this case still wants to connect.

For example:

Following states allow the establishment of a new connection:
- NETWORK_CHANNEL_STATE_OPEN
- NETWORK_CHANNEL_STATE_CONNECTION_IN_PROGRESS
- NETWORK_CHANNEL_STATE_CONNECTED
- NETWORK_CHANNEL_STATE_DISCONNECTED

In contrast these states do not allow the establishment of a connection:
- NETWORK_CHANNEL_STATE_UNINITIALIZED
- NETWORK_CHANNEL_STATE_CLOSED